komodo/prettier.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import print_function
+
+import re
+import sys
+import argparse
+import functools
+
+# On Python3, StringIO can come from standard library io:
+from ruamel.yaml.compat import StringIO
+import ruamel.yaml
+
+
+def repository_specific_formatting(empty_line_top_level, yaml_string):
+    '''Transform function to ruamel.yaml's dump function. Makes sure there are
+    only empty lines inbetween different top level keys (if empty_line_top_level
+    is True, otherwise no empty lines).
+    '''
+
+    yaml_string = re.sub(r'\n+', r'\n', yaml_string)  # Remove all empty lines
+
+    if empty_line_top_level:
+        yaml_string = re.sub(  # Add one empty line between each package
+            r'(\n[^\#][^\n]*)\n([^\s])', r'\1\n\n\2', yaml_string
+        )
+
+    return yaml_string
+
+
+def is_repository(config):
+    '''Returns `False` if the configuration corresponds to a Komodo release
+    (all config elements below top level key are strings). Returns `True` if
+    it corresponds to a _repository_ (all config elements below top level key are
+    themselevs dictionaries).
+
+    Raises ValueError if inconsistent throughout the config.
+    '''
+
+    # For Python2+3 compatibility. On Python3-only, use isinstance(x, str)
+    # instead of isinstance(x, basestring).
+    try:
+        basestring
+    except NameError:
+        basestring = str # No basestring on Python3
+
+    if all([isinstance(package, basestring) for package in config.values()]):
+        return False
+    elif all([isinstance(package, ruamel.yaml.comments.CommentedMap)
+              for package in config.values()]):
+        return True
+
+    raise ValueError(
+        'Inconsistent configuration file. '
+        'Not able to detect if it is a release or repository.'
+    )
+
+
+def prettier(yaml_input_string):
+    '''Takes in a string corresponding to a YAML Komodo configuration, and returns
+    the corresponding prettified YAML string.'''
+
+    ruamel_instance = ruamel.yaml.YAML()
+    ruamel_instance.indent(  # Komodo prefers two space indendation
+        mapping=2, sequence=4, offset=2
+    )
+    ruamel_instance.width = 1000  # Avoid ruamel wrapping long
+
+    config = ruamel_instance.load(yaml_input_string)
```

Comment:
We should try to be concrete on which kind of errors we expect and by reraising a systemexit with the error message the output is much more precise and informative for the user.

```
try:
    config = ruamel_instance.load(yaml_input_string)
except ruamel.yaml.constructor.DuplicateKeyError as e:
    raise SystemExit(str(e))
```

Reply:
Agree - easier (and shorter) user feedback on duplicate keys increases user friendliness. Added suggestion 👍